Hi F'jord!

underscorenygren opened the pull request:
Help printing didn't work for me, not sure if its because of different optparse versions. Added an explicit call to the help function to make it work. 

Never coded ruby before, it sure does look nice.

Commit:
Made help output work for me

Printed out whole object. Added puts so newlines came out right and a .help to print the help fn value

Commit:
Added code to work for ubuntu 12.04.

kristjan commented:
Cool, looks reasonable. Some things before I merge though:
- Don't need `pp` for `puts`; no need to require it
- The change to `'\n'` doesn't get interpreted as a new line and fails to split the script like we expect.

```
1.9.3p194 :001 > a = '
1.9.3p194 :002'> hi
1.9.3p194 :003'> 
1.9.3p194 :004'> bye
1.9.3p194 :005'> '
 => "\nhi\n\nbye\n" 
1.9.3p194 :006 > a
 => "\nhi\n\nbye\n" 
1.9.3p194 :007 > puts a

hi

bye
 => nil 
1.9.3p194 :008 > a.split '\n'
 => ["\nhi\n\nbye\n"] 
1.9.3p194 :009 > a.split "\n"
 => ["", "hi", "", "bye"] 
```
- Refactor the `set_desktop` code into OSX and Linux versions and pick the proper one to use, rather than trying OSX, failing and inlining Linux code. That'll be sad if someone wants to add Windows, etc.
- Don't print anything during the process to avoid noise in everyone's cron mailbox

Also, tip for pull requests: always send a request from a branch other than `master`. The request gets updated as you commit to the branch it came from, so what started as a simple `--help` fix now contains Linux changes because you've kept working. Makes it harder to separate and test changes.

underscorenygren commented:
Roger, roger and rogers. Yeah, haven't coded in Ruby, so everything was a
megahack just to get it to work. And sorry about the rogue pp, forgot to
remove it post-testing.

Good idea on the cron, didn't think about that.

I'll make sure to branch next time, sorry for the clutter.

Erik Nygren
(650) 796 6434

On Fri, Nov 9, 2012 at 5:51 PM, Kristján Pétursson <notifications@github.com

> wrote:
> 
> Cool, looks reasonable. Some things before I merge though:
> - Don't need pp for puts; no need to require it
> - The change to '\n' doesn't get interpreted as a new line and fails
>   to split the script like we expect.
> 
> 1.9.3p194 :001 > a = '
> 1.9.3p194 :002'> hi
> 1.9.3p194 :003'>
> 1.9.3p194 :004'> bye
> 1.9.3p194 :005'> '
>  => "\nhi\n\nbye\n"
> 1.9.3p194 :006 > a
>  => "\nhi\n\nbye\n"
> 1.9.3p194 :007 > puts a
> 
> hi
> 
> bye
>  => nil
> 1.9.3p194 :008 > a.split '\n'
>  => ["\nhi\n\nbye\n"]
> 1.9.3p194 :009 > a.split "\n"
>  => ["", "hi", "", "bye"]
> - Refactor the set_desktop code into OSX and Linux versions and pick
>   the proper one to use, rather than trying OSX, failing and inlining Linux
>   code. That'll be sad if someone wants to add Windows, etc.
> - Don't print anything during the process to avoid noise in everyone's
>   cron mailbox
> 
> Also, tip for pull requests: always send a request from a branch other
> than master. The request gets updated as you commit to the branch it came
> from, so what started as a simple --help fix now contains Linux changes
> because you've kept working. Makes it harder to separate and test changes.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/kristjan/nice_top/pull/1#issuecomment-10247586.
